client/my-sites/site-settings/navigation-back-button/index.jsx

```diff
@@ -0,0 +1,33 @@
+/** @format */
```

Comment:
This is a generic component, not restricted to use with the site settings, so I think it may make sense for it to live at https://github.com/Automattic/wp-calypso/tree/master/client/components.